CHANGELOG.md

```diff
@@ -21,6 +21,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Error message for tests that should panic but pass
 
+### Cast
+
+#### Added
+
+- `--package` flag to specify scarb package to work with
```

Comment:
To which command? What does it mean "to work with"?

Reply:
Hmm, perhaps you're right. I have added it as global for now, since other things use it indirectly (basically everything that touches cast config, eg accounts). This was because I thought that peraps packages should be able to have their own configs, but I'm not sure now. wydt?

Reply:
I think we shouldn't include it when command does not operate on packages. 

Commands having their own configs is not a problem of foundry, scarb should resolve it based on where command is run from.

Reply:
Wait, I'm talking about cast config ([tools.sncast.proflie])

Reply:
Ok, to move things forward I'll remove such possibility (to define configs per package), as perhaps it indeed does not make much sense for some commands. One, top-level config per workspace should be en ough, especially so that we allow for multiple profiles

Reply:
Agreed this should be investigated, we can look into later and discuss it with scarb team.

Reply:
Can we create an issue for investigating how per package configs work?

Reply:
Wdym by top-level config?

Reply:
yes, this is indeed an issue. So when we have per subcommands --package flag, we'll probably need to check members first, take every tool.sncast config from those packages and then check for default config (without profile name) - at least that's the only sensible solution I can think of. Main drawback of this is one can only have one default (tool.scast) profile per workspace. thoughts?

Reply:
Discussed with @piotmag769 possible solutions to this pickle, and we came up with this flow:

```
- we try to declare/script
    - for workspaces:
        - if there is only one package -> return its castconfig (if it exists, else empty)
        - more than one package -> require `--package` flag, return its castconfig (if it exists, else empty)
    - for single packages:
        - return its castconfig (if it exists, else empty)

- we try to invoke/call/account/whatever else:
    - for workspaces:
        - if there is only one package -> return its castconfig (if it exists, else empty)
        - more than one package -> print out warning that package couldn not be determined, and return an empty castconfig
    - for single packages:
        - return its castconfig (if it exists, else empty)
```

It's not ideal, but allows for using configs from packages for subcommands supporting it, and makes other cases more explicit.

**An alternative approach** would be to revert the `--package only to declare/script` altogether, and make it global again which would allow to pick workspaces (and thus, ccastconfigs) for each subcommand. That would be a cleaner and easier solution, but would mean that (in some cases) we must specify a package to invoke some random contract, which is far from perfect. 

Thoughts? Is the solution 1) perfectly acceptable, should we lean to solution 2) with the alternative approach, or maybe something different altogether?